src/renderer/atlas/shader_ps.hlsl

```diff
@@ -152,10 +141,31 @@ float4 main(float4 pos: SV_Position): SV_Target
             }
         }
     }
-    // Step 3: Lines, but not "under"lines
-    [branch] if (cell.flags & CellFlags_Strikethrough)
+    // Step 2: Lines
     {
-        [flatten] if (cellPos.y >= strikethroughPos.x && cellPos.y < strikethroughPos.y)
+        // What a nice coincidence that we have exactly 8 flags to handle right now!
+        bool4x2 flags = {
+            cell.flags & CellFlags_BorderLeft,
+            cell.flags & CellFlags_BorderTop,
+            cell.flags & CellFlags_BorderRight,
+            cell.flags & CellFlags_BorderBottom,
+            cell.flags & CellFlags_Underline,
+            cell.flags & CellFlags_UnderlineDotted,
+            cell.flags & CellFlags_UnderlineDouble,
+            cell.flags & CellFlags_Strikethrough,
+        };
+        bool4x2 checks = {
+            cellPos < thinLineWidth,
+            (cellSize - cellPos) <= thinLineWidth,
+            // The following <lineWidth checks rely on underflow turning the
+            // uint into a way larger number than any reasonable lineWidth.
+            // That way we don't need to write `y >= lo && y < hi`.
+            (cellPos.y - underlinePos) < underlineWidth,
+            (cellPos.y - underlinePos) < underlineWidth && (viewportPos.x / underlineWidth & 3) == 0,
+            any((cellPos.y - doubleUnderlinePos) < thinLineWidth),
+            (cellPos.y - strikethroughPos) < strikethroughWidth,
+        };

```

Comment:
**WHAT** is this crazy thing? it's very cool.

I'm assuming that it runs every check every time, because to branch on all of them would be more expensive?

Reply:
There are only 6 checks... and I am very confused about how this works

Reply:
oh my god, do the first two expand into two checks each

Reply:
I think you need to write a comment explaining how this .. uh, works

Reply:
> I'm assuming that it runs every check every time, because to branch on all of them would be more expensive?

I didn't benchmark this. I simply picked whatever produced the smallest shader and this is only less than half as large as making everything separate branches.